crux-core/src/crux/node.clj

```diff
@@ -100,27 +117,32 @@
                      (tx/tx-events->tx-ops snapshot object-store tx-events))))
         tx-log-entry)))
 
-  (sync [_ timeout]
+  (sync [this timeout]
+    (assert-node-open this)
     (tx/await-no-consumer-lag indexer (or (and timeout (.toMillis timeout))
                                           (:crux.tx-log/await-tx-timeout options))))
 
-  (sync [_ tx-time timeout]
+  (sync [this tx-time timeout]
+    (assert-node-open this)
     (tx/await-tx-time indexer tx-time (or (and timeout (.toMillis timeout))
                                           (:crux.tx-log/await-tx-timeout options))))
 
   ICruxAsyncIngestAPI
-  (submitTxAsync [_ tx-ops]
+  (submitTxAsync [this tx-ops]
+    (assert-node-open this)
     (db/submit-tx tx-log tx-ops))
 
   backup/INodeBackup
   (write-checkpoint [this {:keys [crux.backup/checkpoint-directory] :as opts}]
+    (assert-node-open this)
     (kv/backup kv-store (io/file checkpoint-directory "kv-store"))
     (when (satisfies? tx-log backup/INodeBackup)
       (backup/write-checkpoint tx-log opts)))
 
   Closeable
   (close [_]
-    (when close-fn (close-fn))))
+    (when (and (not closed?) close-fn) (close-fn))
```

Comment:
`@closed?` ?

FWIW, don't know how prevalent this is in the rest of the Clojure world, but I tend to prefix atoms with `!` so I don't forget to do `@!closed?` - worth it here?

Reply:
I'd say it's likely an orthogonal clean-up - I haven't seen this style before tbh.